### DISCUSSION TITLE
unrecognized DNS provider: cloudflare

### DISCUSSION THREAD

**Discussion by GlebTyazhyolov** [go-acme/lego#1514] [Q&A] [answered]
Sorry if it's a dumb question.

When I run
>lego dnshelp

I get the listing which containing cloudflare provider.

If I'm trying to execute lego using this provider, something like
>lego --email somemail@contoso.com --dns cloudflare --domains test.contoso.com run

I get: 
>unrecognized DNS provider: cloudflare

OS: Linux\Ubuntu
Installed version: lego/focal,now 3.2.0-1 amd64

**Comment by ldez** [comment, marked as answer]
Hello,

We don't handle the deb package, then I don't know what really is the binary used by this package.

The latest version of lego is [v4.5.3](https://github.com/go-acme/lego/releases/tag/v4.5.3)

I recommend using our binaries.

`cloudflare` is supported by lego `v3.2.0`.

https://go-acme.github.io/lego/dns/cloudflare/

**Comment by redtux** [comment]
Even if you don't maintain the Ubuntu packages, it might be of interest to you that the issue persists with `4.1.3-3` (build: `lego version dev linux/amd64`) on Ubuntu 22.04.

The help lists `cloudflare` as supported DNS provider, but when running the following command (with `CLOUDFLARE_DNS_API_TOKEN` set), I get the same error as reported above. (See `lego dnshelp` and `lego dnshelp -c cloudflare`.)

```shell
lego --email $ACME_EMAIL --dns cloudflare --domains $DOMAIN run
2022/09/12 01:21:52 unrecognized DNS provider: cloudflare
```

**Reply by ldez** [reply]
The latest version of lego is [v4.8.0](https://github.com/go-acme/lego/releases/tag/v4.8.0)

The deb package seems completely outdated and I don't know what is the real content of this package.

I recommend using our binaries.
